src/main/java/com/solace/samples/java/patterns/DirectReplierNonBlocking.java

```diff
@@ -0,0 +1,122 @@
+        requestReplyMessageReceiver.start();
+        //5-A. Set up an event handler for situations where the reply message could not be published.
+        requestReplyMessageReceiver.setReplyFailureListener(failedReceiveEvent -> System.out.println("### FAILED RECEIVE EVENT " + failedReceiveEvent));
+
+        //6. Create an OutboundMessageBuilder for building the outbound reply message
+        final OutboundMessageBuilder outboundMessageBuilder = messagingService.messageBuilder();
+
+        //7. Define the handler for the incoming message.
+        final RequestReplyMessageReceiver.RequestMessageHandler messageHandler = (inboundMessage, replier) -> {
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+//          System.out.println("The inbound message is : " + inboundMessage.dump()); // Enable this for learning purposes as it logs a String representation of the whole Message
+            final String stringPayload = inboundMessage.getPayloadAsString();
+            System.out.println("The converted message payload is : " + stringPayload);
+
+            //7-B. Create the outbound message payload.
+            final StringBuilder outboundMessageStringPayload = new StringBuilder().append("Hello! Here is a response to your message on topic : ").append(inboundMessage.getDestinationName()).append(" with incoming payload :").append(stringPayload);
+
+            //7-C. Create the outbound message with headers and payload.
+            final OutboundMessage outboundMessage = outboundMessageBuilder.build(outboundMessageStringPayload.toString());
+
+            //This SOP is just for demo purposes, ideally considering the slow nature of console I/O, any such action should be avoided in message processing
+            System.out.println("The outbound message is : " + outboundMessage.getPayloadAsString());
+
+            //7-D. Post the reply to the incoming message
+            replier.reply(outboundMessage);
+        };
+
+        //8. Loop to identify message discards or errors and terminate if required. This should be handled in a more resilient manner
+        System.out.println(API + " " + SAMPLE_NAME + " connected, and running.");
+        //8-A. Create a scheduler variable with the required configurations to process the message in
+        final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(2);
+
+        //Request to register an asynchronous request message handler using supplied thread executor for callbacks. This method represents push-based non-blocking interface.
+        requestReplyMessageReceiver.receiveAsync(messageHandler, executorService);
+
+        while (System.in.available() == 0 && !isShutdown) {
+            if (hasDetectedDiscard) {
```

Comment:
where can `hasDetectedDiscard` get set to `true` ?

Reply:
This should be in the Message handler as per the DirectSubscriber example

https://github.com/SolaceSamples/solace-samples-java/blob/main/src/main/java/com/solace/samples/java/patterns/DirectSubscriber.java#L94-L103

Note that you will have to make the change in the DirectReplierBlocking as well since the `hasDetectedDiscard` is not set to true either

Reply:
So the discardDetection in the DirectReplierBlocking was removed as a comment from @aaron-613 on that PR.
I think I already created this sample before removing that functionality so this unnecesary check remained.
Lets resolve if we want to have discardDetection or not and then I can remove this entirely or put it back in both the places.

Reply:
I have removed the hasDetectedDiscard flag from both the blocking and non-blocking receiver.